daemon/container_operations.go

```diff
@@ -597,7 +597,10 @@ func cleanOperationalData(es *network.EndpointSettings) {
 }
 
 func (daemon *Daemon) updateNetworkConfig(container *container.Container, n libnetwork.Network, endpointConfig *networktypes.EndpointSettings, updateSettings bool) error {
-	if !containertypes.NetworkMode(n.Name()).IsUserDefined() {
+	endpointConfig.DNS = daemon.configStore.DNS
+
+	if !container.EnableServiceDiscoveryOnDefaultNetwork() &&
```

Comment:
wanted to add short id in the default network aswell. But that is not required I guess. Removing